var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -97,31 +96,41 @@ class Llvm(CMakePackage, CudaPackage):
         default=False,
         description="Include debugging code in OpenMP runtime libraries",
     )
-    variant("lldb", default=True, when="+clang", description="Build the LLVM debugger")
+    variant("lldb", default=True, description="Build the LLVM debugger")
     variant("lld", default=True, description="Build the LLVM linker")
     variant("mlir", default=False, when="@10:", description="Build with MLIR support")
     variant(
-        "internal_unwind",
-        default=True,
-        when="+clang",
-        description="Build the libcxxabi libunwind",
+        "libunwind",
+        values=("none", "project", conditional("runtime", when="+clang @6:")),
+        default="runtime",
```

Comment:
Is there ever a reason *not* to build it as a runtime in versions after it becomes safe to do so?

Reply:
Good question. Do you mean that we could have `+libunwind` (or even `+internal_unwind`) and decide on how to build the component based on the version we build and whether `+clang` is set? I thought of it and have to admit that I don't have bulletproof arguments against that.

However, I generally try not to limit the users if possible: if a package can be built in a certain configuration and its developers do not say "don't do that", I don't want to make the decision for the users. Also, `llvm` is heavily used as just a set of libraries (without `clang`) and the users might want them to be built with the same compiler they use to compile their projects.

Reply:
Two more things here:
1. For me, `+clang libunwind=runtime` and `~clang libunwind=project` read better then just `+clang` and `~clang` because I would have to read `package.py` to understand whether the library that I need is built with the compiler that I want to use.
2. If we are talking about some backwards compatibility, all packages that had `llvm+internal_unwind` (`llvm~internal_unwind`) as a dependency can simply go for `llvm libunwind=project` (`llvm libunwind=none`) and don't be afraid that suddenly `libunwind` is built as a runtime.

Reply:
I think that I follow the variant name change. Its just for consistently with upstream and I think this is a win for us.